pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -267,9 +267,12 @@ func (s *Service) getAPIServerClassicELBSpec() (*infrav1.ClassicELB, error) {
 	})
 
 	// The load balancer APIs require us to only attach one subnet for each AZ.
-	subnets := s.scope.Subnets().FilterPrivate()
```

Comment:
Instead of removing this, you can change line 272 to `subnets = s.scope.Subnets().FilterPublic()`.

Thanks for catching this!

Reply:
Thanks for the review @vincepri .  As soon as I submitted the PR I thought you might prefer NOT having the `else` statement.   I've updated and force pushed.